src/Tempest/View/src/Elements/ViewComponentElement.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\View\Elements;
+
+use function Tempest\Support\str;
+use Tempest\View\Element;
+use Tempest\View\Renderers\TempestViewCompiler;
+use Tempest\View\ViewComponent;
+
+final class ViewComponentElement implements Element
+{
+    use IsElement;
+
+    public function __construct(
+        private readonly TempestViewCompiler $compiler,
+        private readonly ViewComponent $viewComponent,
+        array $attributes,
+    ) {
+        $this->attributes = $attributes;
+    }
+
+    public function getViewComponent(): ViewComponent
+    {
+        return $this->viewComponent;
+    }
+
+    public function getSlot(string $name = 'slot'): ?Element
+    {
+        foreach ($this->getChildren() as $child) {
+            if (! $child instanceof SlotElement) {
+                continue;
+            }
+
+            if ($child->matches($name)) {
+                return $child;
+            }
+        }
+
+        if ($name === 'slot') {
+            $elements = [];
+
+            foreach ($this->getChildren() as $child) {
+                if ($child instanceof SlotElement) {
+                    continue;
+                }
+
+                $elements[] = $child;
+            }
+
+            return new CollectionElement($elements);
+        }
+
+        return null;
+    }
+
+    public function compile(): string
+    {
+        $compiled = str($this->viewComponent->compile($this))
+            // Compile slots
+            ->replaceRegex(
+                regex: '/<x-slot\s*(name="(?<name>\w+)")?((\s*\/>)|><\/x-slot>)/',
+                replace: function ($matches) {
+                    $name = $matches['name'] ?: 'slot';
+
+                    $slot = $this->getSlot($name);
+
+                    if ($slot === null) {
+                        return $matches[0];
+                    }
+
+                    return $slot->compile();
+                },
+            )
+
+            // Compile includes, matches self-closing and non-self-closing tags starting with `x-`, including attributes
+            ->replaceRegex(
```

Comment:
This regex doesn't take nested view components into account that have content themselves (via default or named slots)